compiler/bcq-tools/generate_bcq_output_arrays

```diff
@@ -87,7 +87,154 @@ def find_bcq_version(flags):
 
 
 def print_bcqinfo_output_arrays_v1(flags):
-    pass
+    """
+    This function generates a file which includes output arrays of BCQ v1
+    information bundles. Each bundle is consisted with one of candidate
+    operations (BCQ may be applied) and BCQ constant nodes related with
+    the operation.
+    """
+    graph = load_graph(flags.input_path)
+    graph_def = graph.as_graph_def()
+    ops = graph.get_operations()
+
+    # If there is a constant node named PREFIX_1/bcqinfo_alpha,
+    # it is used for applying BCQ to constant node named PREFIX_1.
+    # Collected prefixes will be used for connecting
+    # bcqinfo nodes and prefix node user operations.
+    prefix_set = set()
+    has_dequant_weight = False
+    for op in ops:
+        if op.type == "Const" and "/bcqinfo_" in op.outputs[0].name:
+            # Metadata do not have prefix
+            if "one_compiler/bcqinfo_one_metadata" in op.outputs[0].name:
+                continue
+
+            prefix_index = op.outputs[0].name.index("/bcqinfo_")
+            prefix = op.outputs[0].name[:prefix_index]
+            prefix_set.add(prefix)
+
+            infoname = op.outputs[0].name[prefix_index + 1:-2]
+            if infoname == "bcqinfo_dequant_weight":
+                has_dequant_weight = True
+
+    # Ideal situation is that the users of BCQ applicable constant nodes
+    # are BCQ applicable operations such as MatMul, GatherV2, etc.
+    # However, operations which do not change original values such as
+    # Ideneity or Transpose can exist between them. In view of TensorFlow Lite,
+    # real users of BCQ applicable constant nodes must be found first.
+    # This work is done by BFS search with queue.
+
+    prefix_node_dict = {}  # key : prefix / value : list of candidates
+    matmul_node_prefix_dict = {}  # key : Name of MatMul node / value : prefix
+
+    queue_prefix = list(prefix_set)
+    queue_nodename = [queue_prefix[idx] + ":0" for idx in range(len(queue_prefix))]
```

Comment:
Q) I'm not good at python... what happens if `prefix_set` is empty? is `queue_prefix[idx]` valid?

Reply:
If `prefix_set` is empty, `len(queue_prefix)` becomes 0 and thus for-loop will not be executed :)